libs/ui/lib/resource-meter/ResourceMeter.tsx

```diff
@@ -0,0 +1,48 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, you can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * Copyright Oxide Computer Company
+ */
+
+import { Tooltip } from '../tooltip/Tooltip'
+
+type ResourceMeterProps = {
+  value: number
+  warningThreshold?: number
+  errorThreshold?: number
+}
+
+/**
+ * Show "percent used" relative to "total available"
+ *
+ * This is a vertical indicator showing how a resource's usage compares with
+ * the total amount of the resource available. Note that the current
+ * configuration of this component has "low usage" showing as "positive" (green)
+ * with warning (yellow) and error (red) thresholds at "high usage" levels.
+ */
+export const ResourceMeter = ({
+  value,
+  warningThreshold = 50,
+  errorThreshold = 75,
+}: ResourceMeterProps) => {
+  const isError = errorThreshold && value > errorThreshold
+  const isWarning = !isError && warningThreshold && value > warningThreshold
+  const usagePercent = `${value}%`
+  const label = `${usagePercent} used`
+  const bg = isError ? 'bg-destructive' : isWarning ? 'bg-notice' : 'bg-success'
```

Comment:
Because of the default value, `errorThreshold &&` is redundant unless you want to short-circuit the check if the threshold is zero. And I get the goal of using variable names as a kind of documentation, but the `!isError` bit means you've written this `if` twice: once to get the boolean indicators, and again to convert those indicators into classes.

```ts
const bg =
  value > errorThreshold
    ? 'bg-destructive' // error
    : value > warningThreshold
    ? 'bg-notice' // warning
    : 'bg-success'
```

This is maybe a little ugly, but I like that you don't have to wonder whether `isWarning` and `isError` are going to be used for anything else. We almost never use `// prettier-ignore`, but I'm tempted to do it this:

```ts
// prettier-ignore
const bg =
  value > errorThreshold ? 'bg-destructive'
  : value > warningThreshold ? 'bg-notice'
  : 'bg-success'
```

Reply:
Wow, I swear I hadn't seen this. Prettier is doing a whole thing about this. They're considering formatting nested ternaries in the way I suggested.

https://prettier.io/blog/2023/11/13/curious-ternaries

Reply:
Is this caught by Tailwind JIT? In practice not a big issue since these styles are used so much everywhere else there's little risk of them being omitted, but just a note.

Reply:
I'd also consider `noticeThreshold` and `warningThreshold`  instead – since the only real threshold for error is when you completely run out of capacity.

Reply:
Tailwind looks for strings anywhere in the file.

> The way Tailwind scans your source code for classes is intentionally very simple — we don’t actually parse or execute any of your code in the language it’s written in, we just use regular expressions to extract every string that could possibly be a class name.
>
> We don’t just limit our search to class="..." attributes because you could be using classes anywhere

https://tailwindcss.com/docs/content-configuration#class-detection-in-depth